File: ruby/game.rb
require_relative 'board'

[0, 10, 14, 1, 2, 6, 11, 13, 9, 4, 7, 8, 3, 5, 12].each do |peg|

  start = Time.now
  # Start with the 0 peg being open
  board = Board.new(peg)

  def play(board)
    if board.moves.any?
      count = 0
      board.moves.each do |move|
        board.move(move[0], move[1], move[2])
        count += play(board)
        board.rewind(move[0], move[1], move[2])
      end
      count
    else
      if board.remaining == 1
        # puts "## Begin Solution ##"
        # puts board.history
        # puts "## End Solution ##"
        1
      else
        0
      end
    end

  end

  puts "There are #{play(board)} solutions for opening peg: #{peg} (took #{Time.now - start} seconds)"

end

File: python/game.py
import board
import time

for peg in [0, 10, 14, 1, 2, 6, 11, 13, 9, 4, 7, 8, 3, 5, 12]:

  brd = board.Board(peg)
  start = time.time()

  def play(brd):

    if(len(brd.moves()) > 0):
      count = 0
      for move in brd.moves():
        brd.move(move[0], move[1], move[2])
        count += play(brd)
        brd.rewind(move[0], move[1], move[2])
      return count
    else:
      if(brd.remaining == 1):
        # print("## Begin Solution ##")
        # brd.print_history()
        # print("## End Solution ##")
        return 1
      else:
        return 0

  print("There are {0} solutions for opening peg: {1} (took {2} seconds)".format(play(brd), peg, time.time() - start))
